### ISSUE TITLE
[caffe2] Is there any method to implement learning rate scheduler?

### ISSUE THREAD

**Issue by BIGBALLON** [pytorch/pytorch#8937]
Hello, everyone. 

since tensorflow or pytorch can adjust learning rate easily by using a scheduler function.

```python
epochs = 200
def step_decay_scheduler(epoch):
    if epoch < 81:
        return 0.1
    if epoch < 122:
        return 0.01
    return 0.001

def step_decay_scheduler2(epoch):
    if epoch < 100:
        return 0.1
    if epoch < 150:
        return 0.01
    return 0.001

def cos_scheduler(epoch):
    return (start_lr+end_lr)/2.+(start_lr-end_lr)/2.*math.cos(math.pi/2.0*(epoch/(epochs/2.0)))
```

Is there anyway to use a different learning rate scheduler(e.g. cos scheduler)???

**Comment by ashay-sriv-06** [comment]
Yes, you can define your own rate scheduler as caffe2 seems to lack a cosine scheduler (although you can make use of the LearningRateSchedule). We can implement it in the following manner: 

```python
import math
import numpy as np
from caffe2.python import workspace

def cos_scheduler(epoch, start_lr=0.1, end_lr=0.001, total_epochs=200):
    return (start_lr + end_lr) / 2 + (start_lr - end_lr) / 2 * math.cos(math.pi * epoch / total_epochs)

for epoch in range(200):
    lr = cos_scheduler(epoch)
    workspace.FeedBlob("lr", np.array([lr], dtype=np.float32))
    print(f"Epoch {epoch}, LR={lr}")
```
